homeassistant/components/opentherm_gw/__init__.py

```diff
@@ -94,6 +96,13 @@ async def async_setup_entry(hass, config_entry):
     gateway = OpenThermGatewayDevice(hass, config_entry)
     hass.data[DATA_OPENTHERM_GW][DATA_GATEWAYS][config_entry.data[CONF_ID]] = gateway
 
+    if config_entry.options[CONF_PRECISION]:
+        migrate_options = {}
+        migrate_options[CONF_READ_PRECISION] = config_entry.options[CONF_PRECISION]
+        migrate_options[CONF_SET_PRECISION] = config_entry.options[CONF_PRECISION]
+        migrate_options[CONF_FLOOR_TEMP] = config_entry.options[CONF_FLOOR_TEMP]
```

Comment:
I see what you mean. But in your example `migrate_options` will be `Null` because `.update()` does not return anything
So it will have to be either:
```
migrate_options = dict(config_entry.options)
migrate_options.update(
    {
        CONF_READ_PRECISION: config_entry.options[CONF_PRECISION],
        CONF_SET_PRECISION: config_entry.options[CONF_PRECISION],
    }
)
del migrate_options[CONF_PRECISION]
```

or something in between like:
```
migrate_options = dict(config_entry.options)
migrate_options[CONF_READ_PRECISION] = config_entry.options[CONF_PRECISION]
migrate_options[CONF_SET_PRECISION] = config_entry.options[CONF_PRECISION]
del migrate_options[CONF_PRECISION]
```

What whould you prefer?

Reply:
My bad, that's what I get for typing suggestions without testing...
To me the first solution looks cleaner, but others may disagree.

Reply:
No problem, totally understandable. I agree, will go for the first one then.
Now only have to find a way to test the migration :-)